tests/test_json.py

```diff
@@ -325,6 +325,76 @@ class Config:
     )
 
 
+def test_nested_models_with_inner_encoders():
```

Comment:
please can you split this into multiple tests, you can move some of the model definitions out of the test function(s) to avoid duplication.

Reply:
I split this into a class of tests, because it seemed neater and made sense to me to group the related tests. I did notice that the tests are normally written in quite a flat structure, however. Let me know if I should just convert them to functions.

Reply:
Best if tests are just functions. Class based tests too easily end in a mess.